imgui.cpp

```diff
@@ -3346,11 +3346,25 @@ void ImGui::DestroyContext(ImGuiContext* ctx)
 }
 
 // No specific ordering/dependency support, will see as needed
-void ImGui::AddContextHook(ImGuiContext* ctx, const ImGuiContextHook* hook)
+ImHookID ImGui::AddContextHook(ImGuiContext* ctx, const ImGuiContextHook* hook)
 {
     ImGuiContext& g = *ctx;
     IM_ASSERT(hook->Callback != NULL);
+    IM_ASSERT(hook->Type != ImGuiContextHookType_PendingRemoval);
+    
     g.Hooks.push_back(*hook);
```

Comment:
Also forgot to test for `hook->HookId == 0` as in last comment.

Reply:
Ok, thought you only meant on removal. Will add the check to prevent double add.

Reply:
This means that the User *must* now set HookId to 0 before adding, since it is not init in the structure.